grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -328,6 +329,31 @@ class QueryService implements Status {
         results as Collection<ConsentCollectionLink>
     }
 
+    @SuppressWarnings(["GroovyAssignabilityCheck"])
+    Map<ConsentCollectionLinkDTO, List<StorageDocument>> findCollectionLinksByConsentKeyAndProjectKey(String consentKey, String projectKey) {
+        Map<ConsentCollectionLink, List<StorageDocument>> sampleInfo = new HashMap<>()
+        final session = sessionFactory.currentSession
+        final String query =
+                ' select c.id id, c.consent_key consentKey, c.project_key projectKey, c.pii pii, c.compliance compliance, c.sharing_type sharingType , c.text_sharing_type textSharingType,  ' +
+                        ' c.text_compliance textCompliance, c.require_mta requireMta, c.sample_collection_id sampleCollectionId, ' +
+                        ' sc.name collectionName, ic.summary consentName, ip.summary projectName, c.international_cohorts internationalCohorts ' +
+                        ' from consent_collection_link c ' +
+                        ' inner join issue ic on ic.project_key = c.consent_key ' +
+                        ' inner join issue ip on ip.project_key = c.project_key ' +
+                        ' left join sample_collection sc on sc.collection_id = c.sample_collection_id' +
+                        ' where c.project_key = :projectKey' +
+                        ' and c.consent_key = :consentKey'
+        List<ConsentCollectionLinkDTO> results = session.createSQLQuery(query)
+                .setResultTransformer(Transformers.aliasToBean(ConsentCollectionLinkDTO.class))
```

Comment:
I don't think I've ever used `Transformers. aliasToBean` before. Not sure how that compares to how we do it elsewhere in this class.

Reply:
There wasn't any existent representative object that we could use to be returned to the ui. We considered that in order to not add unnecessary or trash info to a domain object, we thought it would be better a DTO object.

On the other hand, since we weren't using a domain entity, we needed to map somehow what we got from the query to the new DTO object, so as result from a little research, we hit to this solution(`Transformers. aliasToBean`), that will map any given property passed as an alias. 

Does this make sense to you? Please let me know if it needs further explanation.

Reply:
Sure ... works for me.